packages/core/react/src/WalletProvider.tsx

```diff
@@ -235,11 +240,13 @@ export const WalletProvider: FC<WalletProviderProps> = ({
             adapter.on('ready', onReady);
             adapter.on('connect', onConnect);
             adapter.on('disconnect', onDisconnect);
+            adapter.on('onAccountChange', onAccountChanged);
```

Comment:
This naming convention should probably be `change` for consistency. How to implement the change or what changes can be up to the adapter, and then it's more general -- this component will update it's state from anything in the adapter that changed.